pkg/api/helpers.go

```diff
@@ -73,3 +74,11 @@ var standardResources = util.NewStringSet(
 func IsStandardResourceName(str string) bool {
 	return standardResources.Has(str)
 }
+
+func ProbeTimeout(container Container) time.Duration {
```

Comment:
Why take a probe?  Why not just take a Duration and check for 0.

This function is of dubious value, though - by the time kubelet gets it, the api-version-specific default should have been set.  @yujuhong was just refactoring around defaults.